gnoland/website/pkgs/doc/ast_to_string.go

```diff
@@ -0,0 +1,170 @@
+package doc
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+)
+
+func generateFuncSignature(fn *ast.FuncDecl) string {
+	if fn == nil {
+		return ""
+	}
+
+	var b strings.Builder
+	b.WriteString("func ")
+
+	if fn.Recv != nil {
+		var receiverNames []string
+		for _, field := range fn.Recv.List {
+			var fieldName string
+			if len(field.Names) > 0 {
+				fieldName = field.Names[0].Name
+			}
+			receiverNames = append(receiverNames, fmt.Sprintf("%s %s", fieldName, typeString(field.Type)))
+		}
+		if len(receiverNames) > 0 {
+			b.WriteString(fmt.Sprintf("(%s) ", strings.Join(receiverNames, ", ")))
+		}
+	}
+
+	b.WriteString(fmt.Sprintf("%s(", fn.Name.Name))
+
+	var params []string
+	if fn.Type.Params != nil {
+		for _, param := range fn.Type.Params.List {
+			paramType := typeString(param.Type)
+			if len(param.Names) == 0 {
+				params = append(params, paramType)
+			} else {
+				paramNames := make([]string, len(param.Names))
+				for i, name := range param.Names {
+					paramNames[i] = name.Name
+				}
+				params = append(params, fmt.Sprintf("%s %s", strings.Join(paramNames, ", "), paramType))
+			}
+		}
+	}
+
+	b.WriteString(fmt.Sprintf("%s)", strings.Join(params, ", ")))
+
+	results := []string{}
+	if fn.Type.Results != nil {
+		hasNamedParams := false
+		for _, result := range fn.Type.Results.List {
+			if len(result.Names) == 0 {
+				results = append(results, typeString(result.Type))
+			} else {
+				hasNamedParams = true
+				var resultNames []string
+				for _, id := range result.Names {
+					resultNames = append(resultNames, id.Name)
+				}
+				results = append(results, fmt.Sprintf("%s %s", strings.Join(resultNames, ", "), typeString(result.Type)))
+			}
+		}
+
+		if len(results) > 0 {
+			b.WriteString(" ")
+			returnType := strings.Join(results, ", ")
+
+			hasMultipleResultsOrNamedParams := hasNamedParams || len(results) > 1
+			if hasMultipleResultsOrNamedParams {
```

Comment:
```suggestion
			if hasNamedParams || len(results) > 1 {
```

I think it's more readable without the huge variable name :)